src/main/java/com/brave/playlist/playback_service/VideoPlaybackService.kt

```diff
@@ -0,0 +1,223 @@
+import androidx.media3.session.MediaLibraryService
+import androidx.media3.session.MediaSession
+import com.brave.playlist.local_database.PlaylistRepository
+import com.brave.playlist.model.LastPlayedPositionModel
+import com.brave.playlist.model.PlaylistItemModel
+import com.brave.playlist.util.ConstantUtils
+import com.brave.playlist.util.MediaItemUtil
+import com.brave.playlist.util.PlaylistPreferenceUtils
+import com.brave.playlist.util.PlaylistPreferenceUtils.continuousListening
+import com.brave.playlist.util.PlaylistPreferenceUtils.rememberFilePlaybackPosition
+import com.brave.playlist.util.PlaylistPreferenceUtils.rememberListPlaybackPosition
+import com.brave.playlist.util.PlaylistPreferenceUtils.setLatestPlaylistItem
+import com.brave.playlist.util.PlaylistUtils
+import com.google.common.util.concurrent.Futures
+import com.google.common.util.concurrent.ListenableFuture
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+@UnstableApi
+class VideoPlaybackService : MediaLibraryService(),
+    MediaLibraryService.MediaLibrarySession.Callback, Player.Listener {
+    private lateinit var mMediaLibrarySession: MediaLibrarySession
+    private val mScope = CoroutineScope(Job() + Dispatchers.IO)
+
+    private val mPlaylistRepository: PlaylistRepository by lazy {
+        PlaylistRepository(applicationContext)
+    }
+
+    companion object {
+        private lateinit var mPlayer: ExoPlayer
+        var currentPlaylistId: String = ""
+
+        private val mutableCurrentPlayingItem = MutableLiveData<String>()
+        val currentPlayingItem: LiveData<String> get() = mutableCurrentPlayingItem
+        private fun setCurrentPlayingItem(currentPlayingItemId: String) {
+            mutableCurrentPlayingItem.value = currentPlayingItemId
+        }
+
+        private val mutableNewPlaylistItemModel = MutableLiveData<PlaylistItemModel>()
+        val newPlaylistItemModel: LiveData<PlaylistItemModel> get() = mutableNewPlaylistItemModel
+
+        @Suppress("unused")
+        fun addNewPlaylistItemModel(newPlaylistItemModel: PlaylistItemModel) {
+            mutableNewPlaylistItemModel.value = newPlaylistItemModel
+            if (newPlaylistItemModel.playlistId == currentPlaylistId) {
+                val mediaItem = MediaItemUtil.buildMediaItem(
+                    newPlaylistItemModel,
+                    newPlaylistItemModel.playlistId,
+                    newPlaylistItemModel.name, // TODO update playlist name here
+                )
+                mPlayer.addMediaItem(mediaItem)
+            }
+        }
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        initializeSessionAndPlayer()
+        lastSavedPositionTimer()
+    }
+
+    private fun initializeSessionAndPlayer() {
+        val dataSourceFactory = DataSource.Factory { FileDataSource.Factory().createDataSource() }
```

Comment:
Explicitly mentioned `FileDataSource` as DataSource which only works for File Uri